zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -588,7 +587,7 @@ where
                     internal_ovk(),
                     dfvk.change_address().1,
                     *amount,
-                    MemoBytes::empty(),
+                    change_memo.clone().unwrap_or_else(MemoBytes::empty),
```

Comment:
The change to the definition of `SentTransactionOutput::memo` means that the `sapling_output_meta` tuple below this still doesn't match this fixed line. It needs to also be set to `Some(this_value)` so that it isn't using `None` to mean "the user didn't set a change memo so there's an implicit one", instead of the updated definition of "not applicable to this output pool".

Reply:
Ah, yep, I missed that one.